frontend/modules/bim/bim.sagas.ts

```diff
@@ -41,6 +45,28 @@ export function* fetchMetadata({ teamspace, model, metadataId }) {
 	yield put(BimActions.setIsPending(false));
 }
 
+export function* highlightsAllSimilar({ rules }, colour?) {
+	try {
+		const teamspace = yield select(selectCurrentModelTeamspace);
+		const revision = yield select(selectCurrentRevisionId);
+		const model = yield select(selectCurrentModel);
+		const { data } = yield API.getMeshIDsByQuery(teamspace, model, rules, revision);
+
+		return Promise.all(data.map(({ model: modelId, mesh_ids }) => {
+			return Viewer.highlightObjects(teamspace, modelId, colour, true, true, mesh_ids);
```

Comment:
Highlighting needs to go through the tree saga, otherwise it's going to go out of sync with the tree.

Reply:
Thanks for pointing this out, I tested cases with the tree panel and in fact, it didn't work well. However, using Tree sagas action caused a different issue from the user perspective, so whenever we were calling select similar action from bim panel, bim panel was reloaded with new element selected. From what I've understood it was selecting the last node in the provided nodes array. I added an additional property to tree action to omit this in our case.

Additionally, I added tooltips, as I got the feeling that It wasn't really obvious what these new icons do. 

Feel free if You see anything else that can be improved.